pkg/network/tracer/tracer.go

```diff
@@ -343,8 +343,14 @@ func (t *Tracer) addProcessInfo(c *network.ConnectionStats) {
 	addTag("version", p.Envs["DD_VERSION"])
 	addTag("service", p.Envs["DD_SERVICE"])
 
-	if p.ContainerID != "" {
-		c.ContainerID = &p.ContainerID
+	containerID, ok := p.ContainerIndex.Get().(string)
+	if !ok {
+		log.Error("Could not retrieve containerID")
+		return
+	}
+
+	if containerID != "" {
+		c.ContainerID = &containerID
```

Comment:
I think a better change would be to make `c.ContainerID` also an `intern.Value`, but that can be done as a followup.